lib/ecto/changeset.ex

```diff
@@ -185,7 +185,7 @@ defmodule Ecto.Changeset do
 
       changeset =
         {data, types}
-        |> Ecto.Changeset.cast(params["sign_up"], Map.keys(types))
+        |> Ecto.Changeset.cast(data["sign_up"], Map.keys(types))
```

Comment:
I think in this case `params` was right. The first argument would be the tuple `{data, types}` and the second argument would be params to cast onto the given `data`.

Reply:
Yep this is correct after playing a bit more, I closed the pull request and will open a new one which will focus on clarifying this.